From aaed466ba4cb8b4ea075d77a8a086ec25a4f83c4 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Thu, 5 Jul 2018 18:41:39 +0300
Subject: [PATCH 0796/1200] dts: a7040: align comphy and network configurations

Setup A:
	- Add comment for SGMII

Setup C:
	- Comphy1: USB_HOST0 --> SATA0

Setup D:
	- Disable eth1 (SGMII)
	- Enable SFI on comphy2

Documentation will be updated in a separate patch.

Change-Id: I62e8b940df724d8c04f01c1cb78f8e6518d1ca7b
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/57732
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
---
 arch/arm/dts/armada-7040-db-C.dts | 2 +-
 arch/arm/dts/armada-7040-db-D.dts | 8 ++------
 arch/arm/dts/armada-7040-db.dts   | 2 +-
 3 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/arch/arm/dts/armada-7040-db-C.dts b/arch/arm/dts/armada-7040-db-C.dts
index 500b49f905..e6cafdb7e4 100644
--- a/arch/arm/dts/armada-7040-db-C.dts
+++ b/arch/arm/dts/armada-7040-db-C.dts
@@ -148,7 +148,7 @@
 	};
 
 	phy1 {
-		phy-type = <COMPHY_TYPE_USB3_HOST0>;
+		phy-type = <COMPHY_TYPE_SATA0>;
 	};
 
 	phy2 {
diff --git a/arch/arm/dts/armada-7040-db-D.dts b/arch/arm/dts/armada-7040-db-D.dts
index e07ae3e93b..1580fc2c00 100644
--- a/arch/arm/dts/armada-7040-db-D.dts
+++ b/arch/arm/dts/armada-7040-db-D.dts
@@ -179,9 +179,6 @@
 
 &cp0_mdio {
 	status = "okay";
-	phy0: ethernet-phy@0 {
-		reg = <0>;
-	};
 	phy1: ethernet-phy@1 {
 		reg = <1>;
 	};
@@ -191,10 +188,9 @@
 	status = "okay";
 };
 
-&cp0_eth1 {
+&cp0_eth0 {
 	status = "okay";
-	phy = <&phy0>;
-	phy-mode = "sgmii";
+	phy-mode = "sfi"; /* lane-2 */
 };
 
 &cp0_eth2 {
diff --git a/arch/arm/dts/armada-7040-db.dts b/arch/arm/dts/armada-7040-db.dts
index f70db9e4e0..aafaa2d035 100644
--- a/arch/arm/dts/armada-7040-db.dts
+++ b/arch/arm/dts/armada-7040-db.dts
@@ -257,7 +257,7 @@
 &cp0_eth1 {
 	status = "okay";
 	phy = <&phy0>;
-	phy-mode = "sgmii";
+	phy-mode = "sgmii"; /* lane-0 */
 };
 
 &cp0_eth2 {
-- 
2.22.0

